flow/activities/flowable.go

```diff
@@ -368,7 +370,13 @@ func (a *FlowableActivity) SyncFlow(
 			totalRecordsSynced.Add(syncResponse.NumRecordsSynced)
 			logger.Info("synced records", slog.Int64("numRecordsSynced", syncResponse.NumRecordsSynced),
 				slog.Int64("totalRecordsSynced", totalRecordsSynced.Load()))
-
+			if a.OtelManager != nil {
+				a.OtelManager.Metrics.RecordsSyncedGauge.Record(ctx, syncResponse.NumRecordsSynced, metric.WithAttributeSet(attribute.NewSet(
+					attribute.String(otel_metrics.FlowNameKey, config.FlowJobName),
+					attribute.String(otel_metrics.BatchIdKey, strconv.FormatInt(syncResponse.CurrentSyncBatchID, 10)),
+					attribute.String(otel_metrics.SourcePeerType, fmt.Sprintf("%T", srcConn)),
```

Comment:
in behind %T just uses reflect: https://cs.opensource.google/go/go/+/refs/tags/go1.23.5:src/fmt/print.go;l=699

so if you prefer you could make a function in `shared` that does the same thing & then have more flexibility such as dropping package prefix or something